img/posts/2019-07-tx-types.gnuplot

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/gnuplot
+
+set style line 1 lc rgb '#8b1a0e' pt 4 ps 0.25 lt 1 lw 2
+set style line 2 lc rgb '#5e9c36' pt 4 ps 0.25 lt 1 lw 2
+set style line 3 lc rgb '#0025ad' pt 4 ps 0.25 lt 1 lw 2
+set style line 4 lc rgb '#d95319' pt 4 ps 0.25 lt 1 lw 2
+
+set terminal pngcairo size 800,200 font "Sans,12" transparent
+
+set grid
+set tics nomirror
+unset border
+unset key
+#set key horizontal tmargin Left reverse
+#set samples 1000
+
+set style fill transparent solid 0.5 noborder
+
+set format x '%hk'
+set format y '%h%%'
+
+## P2PKH     2  7
+## P2SH      3  8
+## P2WPKH    4  9
+## P2WSH     5  10
+## Nulldata  6  11
+
+set ytics 2
+set xrange [500:]
+set output './2019-07-tx-types-bech32-percentage.png'
+set xlabel "Block number"
+set ylabel "Bech32 outputs\n "
+plot '~/2019-07-all-tx-types-averaged-10000.data' u ($1/1000):(($9+$10)/($7+$8+$9+$10+$11)*100) w line ls 3, \
```

Comment:
Is this a file that contains the tx data? If so, should it be included in this commit? (without it, this .gnuplot file is less useful)

Reply:
Yeah, that's the data file.  Unfortunately it's 100,000 lines (one for each block) and 6.2 MB, which seemed like a bit too much to add to the repository as a one-off data file.  Happy to add it if you want, though.  I can also add the script I used to generate the data (and the other script I used to average it), though they're currently hackish I-didn't-expect-anyone-to-ever-see-this code.

In the past, I haven't added gnuplot files unless I also added everything needed to execute them.  That has lead to me losing some files and being unable to recreate graphs using the exact same parameters, so this seemed better to me.  However, I'm also willing to remove this file if it's an issue.

Reply:
I agree that we shouldn't burden this repo with a 6MB file.

I'd ACK adding the scripts to `_contrib` and adding a reference to that here, so we/others can reproduce the data if required. That can be done as a separate PR, and only if you agree that it's useful.